src/main/java/org/apache/cassandra/sidecar/utils/HttpExceptions.java

```diff
@@ -54,7 +54,7 @@ public static HttpException cassandraServiceUnavailable()
      */
     public static HttpException wrapHttpException(HttpResponseStatus status, Throwable cause)
     {
-        return wrapHttpException(status, null, cause);
+        return wrapHttpException(status, cause != null ? cause.getMessage() : null, cause);
```

Comment:
I would against passing the cause message as the response payload. The message is internal to server and probably does not make sense to the client, or even not desired to be exposed. 
In the scenario where payload is needed, it should set the response message payload explicitly.

Reply:
makes sense, I have addressed this feedback. Thanks